### PR TITLE
Add audit logs to cluster-logging-log-forwarding enhancement proposal

### PR DESCRIPTION
**Pull request by JAORMX**
This is an addition to the already accepted proposal for log forwarding,
which suggests the addition of the audit logs in a separate pipeline.

These logs will not be forwarded by default.

This comes from the need to store audit logs in a secure location, as
part of custmers needing to meet government regulations.

cc: @jhrozek @mrogers950 @jcantrill @richm @bparees

### CODE REVIEW

File: enhancements/cluster-logging/cluster-logging-log-forwarding.md [hunk 1]
```diff
@@ -63,6 +64,7 @@ It will not allow configuration of additional sources without further design con
 intended to provide a complex routing solution as one might achieve by using a custom collector configuration or a messaging solution (e.g. kafka).
 * It is not a goal for the tech-preview to support log forwarding outputs other then the ones identified for the goals. Admins can forward to their own fluentd via `forward` 
 and then configure that fluentd to forward to any number of specific logstore outputs
+* It is not a goal to provide secure storage for audit logs. If the deployer chooses to enable audit log forwarding, they need to make sure that the endpoint is compliant with governmental regulations and secure.
```
Comment (richm) [inline enhancements/cluster-logging/cluster-logging-log-forwarding.md]: "The OpenShift logging Elasticsearch does not comply with those regulations."

File: enhancements/cluster-logging/cluster-logging-log-forwarding.md [hunk 2]
```diff
@@ -71,10 +73,10 @@ Log forwarding will provide a declarative way to specify the outputs for specifi
 
 * `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
 * `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+* `logs.audit` - Logs generated by the nodes' auditd (/var/log/audit/audit.log), audit logs from the kubeapi-server and the openshift-apiserver. This will not be forwarded by default.
```
Comment (richm) [inline enhancements/cluster-logging/cluster-logging-log-forwarding.md]: Do we want to have 2 or 3 different ones?  1) system audit logs are parsed and treated very differently than openshift/kube logs.  2) there may be customers who want to collect openshift audit logs but not system audit logs, and vice versa

Comment (JAORMX) [reply enhancements/cluster-logging/cluster-logging-log-forwarding.md]: Well, right now the request is to have the audit logs (in general) forwarded to a secure location. If we do get requests from customers to have different pipelines, we could split them up.
* logs.audit for all the audit logs
* logs.linux.audit for the auditd logs
* logs.k8s.audit for the k8s audit logs
* logs.openshift.audit for the openshift audit logs

What do you think? (of separating this in a further iteration)

File: enhancements/cluster-logging/cluster-logging-log-forwarding.md (hunk 1 above)
Comment (JAORMX) [reply enhancements/cluster-logging/cluster-logging-log-forwarding.md]: Done! Thanks!

File: enhancements/cluster-logging/cluster-logging-log-forwarding.md (hunk 2 above)
Comment (richm) [reply enhancements/cluster-logging/cluster-logging-log-forwarding.md]: ok - @sichvoge this is something to put on the roadmap for 4.4

**Pull request merged**